packages/beachball/src/publish/publishToRegistry.ts

```diff
@@ -13,15 +14,7 @@ export async function publishToRegistry(originalBumpInfo: BumpInfo, options: Bea
   const { registry, tag, token, access, timeout } = options;
   const bumpInfo = _.cloneDeep(originalBumpInfo);
   const { modifiedPackages, newPackages, packageInfos } = bumpInfo;
-
-  // Execute prepublish hook if available
-  if (options.hooks?.prepublish) {
-    const maybePromise = options.hooks.prepublish(bumpInfo);
-
-    if (maybePromise instanceof Promise) {
-      await maybePromise;
-    }
-  }
+  const prepublishHook = options.hooks?.prepublish;
```

Comment:
can we preserve the ability to do things with a promise? this way, we give a bit more flexibility to what can be done inside the hook

Reply:
Updated